### PR TITLE
fix: new test tests after removal of new test flow in the Studio welcome screen

### PR DESCRIPTION
**Pull request by mabela416**
- Closes 
- Related to removing the new test button from the welcome screen from this story https://github.com/cypress-io/cypress-services/issues/12261
- https://github.com/cypress-io/cypress-services/issues/12510

### Additional details

We're no longer creating new tests from the welcome screen in studio. Now the only ways to create tests are from the suites or from the spec header menu so I updated the tests related to these changes.

---

> [!NOTE]
> Refactors Studio new-test UX/tests away from the welcome screen and introduces URL state to preserve context.
> 
> - Adds `openNewTestFromSpecHeader()` helper; simplifies `inputNewTestName()`; unskips and adds E2E to create from spec header/empty spec, persist name screen and welcome screen across reloads, and verify URL params
> - Enhances `studio-store` to read/write `entrySource` in URL hash alongside `suiteId/testId/sessionId`, clear it when switching modes, include it in `_updateUrlParams`/`_removeUrlParams`, and set from URL on setup
> - Reporter emits `studio:init:suite` with `{ entrySource: 'new-test-root' }` for root suite; tests updated to expect this
> - Changelog: notes bugfixes for creating new tests and empty spec behavior
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit da6a0fa4eaaf69cdcf1343f3769bd7019f1dfe19. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### Steps to test

Before 
- When we reload a page while the name test screen is open, we don't display it again after reloading and instead display the welcome screen. 

https://github.com/user-attachments/assets/7e9946a9-b890-4339-9961-d15667d639ea

Now
- it should refresh to the name test screen

https://github.com/user-attachments/assets/7a47fc10-c654-42d8-9d7e-b8471d2fd364

Before
When opening an empty spec
   - If we clicked on "Create test with Cypress Studio", it would open the welcome screen instead of the name test screen.

https://github.com/user-attachments/assets/d77dd5e7-d16f-46cd-ad35-a798ade8eec2

   - If clicked on the "Studio" button and then clicked on "Create test with Cypress Studio", it would continue to display the welcome screen instead of displaying the name test screen.

https://github.com/user-attachments/assets/a3ea297e-f6e5-4b88-b26e-6343d169a435

Now 
- Clicking "Create test with Cypress Studio" with or without the welcome screen open, should open the name test screen

https://github.com/user-attachments/assets/5dbf70c2-ccb2-47b7-b086-c03f6a944ffe

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/app/cypress/e2e/studio/studio-new-tests.cy.ts [hunk 1]
```diff
@@ -163,7 +163,9 @@ describe('studio functionality', () => {
   it('creates a new test from an empty spec', () => {
     loadProjectAndRunSpec({ specName: 'empty.cy.js', specSelector: 'title' })
 
-    cy.contains('Create test with Cypress Studio').click()
+    cy.waitForSpecToFinish()
```
Comment (mschile) [inline packages/app/cypress/e2e/studio/studio-new-tests.cy.ts:171]: I believe `loadProjectAndRunSpec` already calls `waitForSpecToFinish`.

**Review by mschile — APPROVED** [review]

File: packages/app/cypress/e2e/studio/studio-new-tests.cy.ts [hunk 2]
```diff
@@ -163,7 +163,9 @@ describe('studio functionality', () => {
   it('creates a new test from an empty spec', () => {
     loadProjectAndRunSpec({ specName: 'empty.cy.js', specSelector: 'title' })
 
-    cy.contains('Create test with Cypress Studio').click()
```
Comment (mschile) [inline packages/app/cypress/e2e/studio/studio-new-tests.cy.ts:167]: I think we do want to click the `Create test with Cypress Studio` link since we want to test creating a new test from that view.

**Review by mschile — CHANGES_REQUESTED** [review]
Added a comment.

File: packages/app/cypress/e2e/studio/studio-new-tests.cy.ts (hunk 2 above)
Comment (mabela416) [reply packages/app/cypress/e2e/studio/studio-new-tests.cy.ts:167]: Restored this and fixed the logic for this link since it wasn't working with the new flow 

https://github.com/cypress-io/cypress/pull/33236/commits/5bd082fdf6e63c07226b6b3c0863faa95f5d2bc6

File: packages/app/cypress/e2e/studio/studio-new-tests.cy.ts (hunk 1 above)
Comment (mabela416) [reply packages/app/cypress/e2e/studio/studio-new-tests.cy.ts:171]: Removed that https://github.com/cypress-io/cypress/pull/33236/commits/5bd082fdf6e63c07226b6b3c0863faa95f5d2bc6

**Review by mschile — DISMISSED** [review]

**Comment by mschile** [comment]
@mabela416, will you please make sure to un-skip any of the tests that reference this PR.

File: packages/app/cypress/e2e/studio/studio-new-tests.cy.ts [hunk 3]
```diff
@@ -160,10 +160,41 @@ describe('studio functionality', () => {
...
+      const spec = await ctx.actions.file.readFileInProject('cypress/e2e/empty.cy.js')
+
+      expect(spec.trim().replace(/\r/g, '')).to.equal(`
+it('new-test', function() {
+    cy.visit("cypress/e2e/index.html")
+});`.trim())
+    })
+  })
+
+  it('opens the name test screen from the welcome screen and creates a new test from an empty spec ', () => {
```
Comment (mabela416) [inline packages/app/cypress/e2e/studio/studio-new-tests.cy.ts]:
```suggestion
  it('opens the name test screen from the welcome screen and creates a new test from an empty spec', () => {
```

File: packages/app/src/store/studio-store.ts [hunk 4]
```diff
@@ -202,6 +203,7 @@ export const useStudioStore = defineStore('studioRecorder', {
       this.testId = undefined
       this.suiteId = undefined
       this.newTestLineNumber = undefined
+      this.entrySource = undefined
```
[the remark this answers is not in the document]
Comment (mabela416) [reply packages/app/src/store/studio-store.ts:208]: fixed here https://github.com/cypress-io/cypress/pull/33236/commits/514813c0564fb591775c290a41e031c87c53b71a

**Review by astone123 — APPROVED** [review]

**Comment by mschile** [comment]
@mabela416, since this PR now contains functional changes, the PR title will need to be updated and possibly an entry into the changelog.

**Review by mschile — APPROVED** [review]
Added one comment but looks good overall!

File: cli/CHANGELOG.md [hunk 5]
```diff
@@ -22,6 +22,9 @@ _Released 01/27/2026 (PENDING)_
 - Upgraded `shell-env` to `4.0.1` and `@cypress/commit-info` to `2.2.2`. This removes the [GMS-2020-2](https://gitlab.com/gitlab-org/security-products/gemnasium-db/-/blob/master/npm/execa/GMS-2020-2.yml) vulnerability being reported in security scans. Addressed in [#33226](https://github.com/cypress-io/cypress/pull/33226) and [#33263](https://github.com/cypress-io/cypress/pull/33263).
 - Upgraded `lodash` to `4.17.23`. This removes the [CVE-2025-13465](https://github.com/advisories/GHSA-xxjr-mmjv-4gpg) vulnerability being reported in security scans. Addresses [#33269](https://github.com/cypress-io/cypress/issues/33269).
 
+**Bugfixes:**
+
+- Fixed an issue where the user did not always have the ability to create a new test in Studio. Also, fixed an issue where creating a new test from an empty spec would display the welcome to studio screen instead of the form to name the new test. Addresses [#12510](https://github.com/cypress-io/cypress-services/issues/12510). Addressed in [#33236](https://github.com/cypress-io/cypress/pull/33236).
```
Comment (mschile) [inline cli/CHANGELOG.md]:
```suggestion
- Fixed an issue where the user did not always have the ability to create a new test in Studio. Also, fixed an issue where creating a new test from an empty spec would display the welcome to studio screen instead of the form to name the new test. Addressed in [#33236](https://github.com/cypress-io/cypress/pull/33236).
```
